src/System.CommandLine.Suggest/dotnet-suggest-shim.bash

```diff
@@ -1,30 +1,30 @@
 # dotnet suggest shell complete script start
 _dotnet_bash_complete()
 {
-    local fullpath=`type -p ${COMP_WORDS[0]}`
-    local escaped_comp_line=$(echo "$COMP_LINE" | sed s/\"/'\\\"'/g)
-    local completions=`dotnet-suggest get --executable "${fullpath}" --position ${COMP_POINT} -- "${escaped_comp_line}"`
+    local fullpath=$(type -p "${COMP_WORDS[0]}")
+    local completions=$(dotnet-suggest get --executable "${fullpath}" --position "${COMP_POINT}" -- "${COMP_LINE}")
 
+    # filter suggestions by word to complete.
+    local word="${COMP_WORDS[COMP_CWORD]}"
     local IFS=$'\n'
-    local suggestions=($(compgen -W "$completions"))
+    local suggestions=($(compgen -W "$completions" -- "$word"))
```

Comment:
This is the fix for https://github.com/dotnet/command-line-api/issues/2143.

Reply:
This portion makes complete sense to me:

* COMP_WORDS - array of parsed words
* COMP_CWORD - array index of the current cursor position
* ${COMP_WORDS[COMP_CWORD]} - current whole word
* `compgen -W "$completions" -- "$word"` - -W is the full list of potential completions, and specifying a `$word` argument tells compgen which words to filter on